week3/d01/Assignment_3_1_1/ex3.cpp

```diff
@@ -0,0 +1,88 @@
+#include <iostream>
+#include <string>
+#include <vector>
+#include <limits>
+#include <exception>
+
+void clear_cin()
+{
+    std::cin.clear();
+    std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+}
+
+struct Names
+{
+    int id {};
+    std::string name {};
+};
+
+void print_names(std::vector<Names>& names)
+{
+    for (Names name : names)
```

Comment:
I'm not a 100% but I think this defeats the point of taking the vector in as reference, as this range based for loop takes names as value, not reference.